app/src/main/java/org/dhis2/data/forms/dataentry/fields/FieldViewModelFactoryImpl.java

```diff
@@ -166,22 +166,29 @@ public FieldUiModel create(@NonNull String id,
                             type
                     );
                 } else if (fieldRendering != null && type == ValueType.TEXT && optionSetTextRenderings.contains(fieldRendering.type())) {
-                    return OptionSetViewModel.create(
+                    return new FieldUiModelImpl(
```

Comment:
This code is outdated with the base branch, please update you working branch.
Also, I would remove refactored cases and gather only one FieldViewModelImpl(...)